### PR TITLE
Add option to enable encryption at host

### PR DESCRIPTION
**Pull request by eetu-n**
Added option to enable encryption at host when provisioning VMs.

https://github.com/jenkinsci/azure-vm-agents-plugin/issues/415

### Testing done
Tested provisioning VMs on a subscription where encryption at host is mandatory.

### Submitter checklist

- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] Link to relevant issues in GitHub or Jira
- [x] Link to relevant pull requests, esp. upstream and downstream changes
- [x] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

File: src/main/java/com/microsoft/azure/vmagent/AzureVMManagementServiceDelegate.java [hunk 1]
```diff
@@ -495,6 +496,7 @@ public AzureVMDeploymentInfo createDeployment(
             copyVariableIfNotBlank(tmp, properties, "imageVersion");
             copyVariableIfNotBlank(tmp, properties, "osType");
             putVariable(tmp, "ephemeralOSDisk", Boolean.toString(ephemeralOSDisk));
+            putVariable(tmp, "encryptionAtHost", Boolean.toString(encryptionAtHost));
```
Comment (timja) [inline src/main/java/com/microsoft/azure/vmagent/AzureVMManagementServiceDelegate.java:499]: adding this variable won't do anything, you need to either modify all the arm templates to add something that uses this variable, or add some code that inserts it into the arm template around here somewhere.

File: src/main/resources/com/microsoft/azure/vmagent/AzureVMAgentTemplate/config.properties [hunk 2]
```diff
@@ -17,6 +17,7 @@ noOfParallelJobs=Number of executors
 Image_Configuration=Image configuration
 Disk_Type=Choose disks type
 Use_Ephemeral_OS_Disk=Use Ephemeral OS disk
+Enable_Encryption_At_Host=Enable Encryption At Host
```
Comment (timja) [inline src/main/resources/com/microsoft/azure/vmagent/AzureVMAgentTemplate/config.properties:20]:
```suggestion
Enable_Encryption_At_Host=Enable Encryption at host
```

**Review by timja — COMMENTED** [review]
Are you sure when you tested it that it had encryption at host enabled?

I can't see anything actually setting it.
apart from that looks good, (untested as of yet)

**Comment by eetu-n** [comment]
Hmm ok, I'll have to do some more testing once I'm done with this different issue with our setup. Might be that the error changing to a different one after my updates was just a coincidence.

**Comment by eetu-n** [comment]
Hi,
so seems I didn't notice nothing had changed as we were on quite an old version of the plugin, and the current version I applied my patch to seems to have done things in a different order, thus I got a different error and assumed I'd fixed the issue. Now I've added the changes to the appropriate arm-templates and the plugin is deploying new nodes correctly.

**Review by timja — APPROVED** [review]
I've tested that it doesn't break anything, I haven't enabled the feature but looks good to me.

Thanks!

**Pull request merged**